### ISSUE TITLE
Getting Time objects from FITS headers

### ISSUE THREAD

**Issue by dstansby** [astropy/astropy#11297]
### Description

Several entries in FITS headers indicate a time, e.g. DATE-OBS, DATE-BEG, DATE-END etc. For more information, see section 9, table 35 of the FITS standard: https://fits.gsfc.nasa.gov/standard40/fits_standard40aa-le.pdf

As far as I can tell, there is no simple way to get these out of an `astropy.io.fits.Header` object, and into a `astropy.time.Time` object. The best I have is
```python
tsring = header['DATE-OBS']
scale = header['TIMESYS']
dtime = Time(tstring, scale=scale, format='fits')
```

It would be nice if there was a way for astropy to automatically apply the scale and any other necessary keywords, e.g. MJDREF. Perhaps a `Header.get_time(key)` method where `key` could be e.g. DATE-OBS would be helpful?

I would be happy to implement such a helper, but thought it would be best to first seek advice on whether this would be welcomed/what the API should be.

### Additional context

There is https://github.com/astropy/astropy/issues/6676 already open, but I think that refers to times stored as data in columns as opposed to the header.

**Comment by saimn** [comment]
There is also #10073 about setting FITS headers with `datetime` or `Time` instances.

**Comment by mhvk** [comment]
Indeed, this would be great! I think it may be best to move discussion to #10073, and expand that to setting and getting. (I'll comment more there.)
